crates/bevy_pbr/src/render/pbr_lighting.wgsl

```diff
@@ -284,3 +285,151 @@ fn directional_light(light_id: u32, roughness: f32, NdotV: f32, normal: vec3<f32
+    let offset_position = (clip_exit_position.xy / clip_exit_position.w) * vec2<f32>(0.5, -0.5) + 0.5;
+
+    // Fetch background color
+    let background_color = fetch_transmissive_background(offset_position, frag_coord, perceptual_roughness, distance);
+
+    // Calculate final color by applying transmissive color to a mix of background color and transmitted specular environment light
+    // TODO: Add support for attenuationColor and attenuationDistance
+    return transmissive_color * mix(transmitted_environment_light_specular, background_color.rgb, background_color.a);
+}
+
+// https://blog.demofox.org/2022/01/01/interleaved-gradient-noise-a-different-kind-of-low-discrepancy-sequence
+fn interleaved_gradient_noise(pixel_coordinates: vec2<f32>) -> f32 {
+#ifdef TEMPORAL_JITTER
+    let frame = f32(view_bindings::globals.frame_count % 64u);
+    let xy = pixel_coordinates + 5.588238 * frame;
+#else
+    // Don't vary noise per frame if TAA is not enabled
+    let xy = pixel_coordinates;
+#endif
+    return fract(52.9829189 * fract(0.06711056 * xy.x + 0.00583715 * xy.y));
+}
+
+// https://www.iryoku.com/next-generation-post-processing-in-call-of-duty-advanced-warfare (slides 120-135)
+const SPIRAL_OFFSETS: array<vec2<f32>, 8> = array<vec2<f32>, 8>(
+    vec2<f32>(-0.7071,  0.7071),
+    vec2<f32>(-0.0000, -0.8750),
+    vec2<f32>( 0.5303,  0.5303),
+    vec2<f32>(-0.6250, -0.0000),
+    vec2<f32>( 0.3536, -0.3536),
+    vec2<f32>(-0.0000,  0.3750),
+    vec2<f32>(-0.1768, -0.1768),
+    vec2<f32>( 0.1250,  0.0000),
+);
+
+const MAX_TRANSMISSIVE_TAPS = 16;
+
+fn fetch_transmissive_background(offset_position: vec2<f32>, frag_coord: vec3<f32>, perceptual_roughness: f32, distance: f32) -> vec4<f32> {
+    // Calculate view aspect ratio, used to scale offset so that it's proportionate
+    let aspect = view_bindings::view.viewport.z / view_bindings::view.viewport.w;
+
+    // Calculate how “blurry” the transmission should be.
+    // Blur is more or less eyeballed to look approximately “right”, since the “correct”
+    // approach would involve projecting many scattered rays and figuring out their individual
+    // exit positions. IRL, light rays can be scattered when entering/exiting a material (due to
+    // roughness) or inside the material (due to subsurface scattering). Here, we only consider
+    // the first scenario.
+    //
+    // Blur intensity is:
+    // - squarely proportional to `perceptual_roughness`
+    // - inversely proportional to object distance to the view
+    let blur_intensity = (perceptual_roughness * perceptual_roughness) / distance;
+
+    // Number of taps scale with blur intensity
+    let num_taps = i32(max(min(sqrt(blur_intensity) * 7.0, 1.0) * f32(MAX_TRANSMISSIVE_TAPS), 1.0));
+    let num_spirals = (num_taps >> 3u) + 1;
+    let random_angle = interleaved_gradient_noise(frag_coord.xy);
+#ifdef TEMPORAL_JITTER
+    // Alternating pixel mesh: 0 or 1 on even/odd pixels, alternates every frame
```

Comment:
What does `pixel_mesh` mean here?

Reply:
Essentially this, it's a way to make the interleaved gradient noise less noticeable, especially without the temporal jitter

![image](https://github.com/bevyengine/bevy/assets/418473/c9f030c6-7d6e-4de6-b922-01056334b8fb)

Maybe I should name it something else, like `pixel_checkerboard`?

Reply:
Yeah checkerboard would be better, and please add this as a comment.